### PR TITLE
Rename p2300 variant with stdexec for pika package

### PR DESCRIPTION
**Pull request by aurianer**
We now rename the variant in pika since the senders/receivers implementation moved from https://github.com/brycelelbach/wg21_p2300_execution to https://github.com/NVIDIA/stdexec.

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Looks like all the associated references to `p2300` in the package were changed and I can't find any built-in packages or CI stacks that explicitly utilize the original variant name.

**Comment by tldahlgren** [comment]
Let me know if you want this merged now or if I should give time for one or more of the other maintainers to review the change.

**Comment by aurianer** [comment]
it's not urgent so we can wait for @msimberg to review it, thank you :)

File: var/spack/repos/builtin/packages/pika/package.py
Comment (msimberg) [inline var/spack/repos/builtin/packages/pika/package.py]:
```suggestion
        description="Use stdexec for sender/receiver functionality",
```

**Review by msimberg — CHANGES_REQUESTED** [review]
Just comment on updating the description of the `stdexec` variant. Looks good otherwise!

**Review by msimberg — APPROVED** [review]

**Pull request merged**
